execution-core/src/bytecode.rs

```diff
@@ -0,0 +1,59 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+//
+// Copyright (c) DUSK NETWORK. All rights reserved.
+
+//! Wrapper for a long data that we want to keep the integrity of.
+
+extern crate alloc;
+use alloc::vec::Vec;
+use bytecheck::CheckBytes;
+use dusk_bytes::Error as BytesError;
+use dusk_bytes::Error::InvalidData;
+use rkyv::{Archive, Deserialize, Serialize};
+
+#[derive(Debug, Clone, PartialEq, Eq, Archive, Serialize, Deserialize)]
+#[archive_attr(derive(CheckBytes))]
+/// Holds bytes of bytecode and its hash.
+pub struct Bytecode {
+    /// Hash of the bytecode bytes.
+    pub hash: [u8; 32],
+    /// Bytecode bytes.
+    pub bytes: Vec<u8>,
+}
```

Comment:
I'm puzzled as to the need for this structure :thinking: . Why would one need to have the bytecode and its hash together?

Reply:
That is because they behave as a unit. Hash represents `bytes` in functions like `to_hash_input_bytes` and in `to_var_bytes`. Alternative would be to put them separately in `ContractDeploy`, but that would make the relation between hash and bytes not explicit, basically hidden from the reader. Since we have to introduce redundancy, better make it explicit IMO. Bytecode is a unit and it is represented as such struct. It is better than having two members of ContractDeploy, unconnected to the eye of the reader, yet in reality, very connected.

Reply:
I took a further look and now think this is actually quite a smart solution for stripping the bytecode from the transaction without using formal typing. Nice. As long as we make sure that the bytecode can't just be swapped by something else - i.e. the network checks the has matches the bytecode - this should be fine.

Reply:
Glad to hear it. Not sure I understand `network checks the has matches the bytecode` - could you explain?